modules/images/webp-uploads/load.php

```diff
@@ -683,3 +683,99 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
+ * @param string $file                  Path to original file.
+ * @return array                        Metadata with sources added.
+ */
+function webp_uploads_update_sources( $metadata, $valid_mime_transforms, $allowed_mimes, $file ) {
+	$original_directory = pathinfo( $file, PATHINFO_DIRNAME );
+
+	foreach ( $metadata['sizes'] as $size => $size_details ) {
+		foreach ( $valid_mime_transforms as $targeted_mime ) {
+			$extension            = explode( '|', $allowed_mimes[ $targeted_mime ] );
+			$filename_without_ext = implode( explode( '.', $size_details['file'], -1 ) );
+			$newfile              = trailingslashit( $original_directory ) . $filename_without_ext . ".{$extension[0]}";
+
+			$metadata['sizes'][ $size ]['sources'][ $targeted_mime ] = array(
+				'file'     => $newfile,
+				'filesize' => filesize( $newfile ),
+			);
+		}
+	}
+
+	return $metadata;
+}
+
+/**
+ *
+ * Creates additional image formats when original image is edited.
+ *
+ * @since n.e.x.t
+ *
+ * @param bool|null       $override  Value to return instead of saving. Default null.
+ * @param string          $file      Name of the file to be saved.
+ * @param WP_Image_Editor $image     The image editor instance.
+ * @param string          $mime_type The mime type of the image.
+ * @param int             $post_id   Attachment post ID.
+ *
+ * @return bool|null
+ */
+function webp_uploads_update_image_onchange( $override, $file, $image, $mime_type, $post_id ) {
+	if ( null !== $override ) {
+		return $override;
+	}
+
+	$valid_mime_transforms = webp_uploads_get_upload_image_mime_transforms()[ $mime_type ];
+	$original_directory    = pathinfo( $file, PATHINFO_DIRNAME );
+	$filename              = pathinfo( $file, PATHINFO_FILENAME );
+	$sizes                 = wp_get_registered_image_subsizes();
+	$allowed_mimes         = array_flip( wp_get_mime_types() );
+
+	foreach ( $valid_mime_transforms as $targeted_mime ) {
+		if ( $targeted_mime === $mime_type ) {
+			continue;
+		}
+
+		if ( ! isset( $allowed_mimes[ $targeted_mime ] ) || ! is_string( $allowed_mimes[ $targeted_mime ] ) ) {
+			return $override;
+		}
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $targeted_mime ) ) ) {
+			return $override;
+		}
+
+		$extension   = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$destination = trailingslashit( $original_directory ) . "{$filename}.{$extension[0]}";
+
+		if ( is_wp_error( $image->save( $destination, $targeted_mime ) ) ) {
+			return $override;
+		}
+
+		$_sizes = array();
+
+		foreach ( $sizes as $size => $size_details ) {
+			$_sizes[ $size ] = $size_details;
+		}
+
+		$image->multi_resize( $_sizes );
+	}
+	add_filter(
+		'wp_update_attachment_metadata',
+		function ( $metadata, $post_meta_id ) use ( $post_id, $valid_mime_transforms, $allowed_mimes, $file ) {
+			if ( $post_meta_id !== $post_id ) {
+				return $metadata;
+			}
+
+			return webp_uploads_update_sources( $metadata, $valid_mime_transforms, $allowed_mimes, $file );
+		},
+		10,
+		2
+	);
```

Comment:
Looks like we are missing the filterable value.

```suggestion
      return $override;
	);
```